pkg/kubectl/sorting_printer.go

```diff
@@ -185,16 +186,37 @@ func isLess(i, j reflect.Value) (bool, error) {
 
 func (r *RuntimeSort) Less(i, j int) bool {
 	iObj := r.objs[i]
+	iElem := reflect.ValueOf(iObj).Elem()
 	jObj := r.objs[j]
+	jElem := reflect.ValueOf(jObj).Elem()
+	if r.field == "" && iElem.Type().Name() == "ReplicaSet" && jElem.Type().Name() == "ReplicaSet" {
```

Comment:
I think this is an argument that "sort" is something that may need to be passed in at a higher level, prior to the execution of these items.  Arguably this should be done server side, rather than client side, where you won't have to cast the data.  I believe doing that sort on the server side should be possible - there is already a sort in place on the returned entries.  Altering the default sort is reasonable, although we'd need to discuss the API compatibility implications.